packages/task/src/browser/task-configurations.ts

```diff
@@ -129,16 +129,14 @@ export class TaskConfigurations implements Disposable {
      *
      * The invalid task configs are not returned.
      */
-    async getTasks(token: number): Promise<TaskConfiguration[]> {
+    async getTasks(): Promise<TaskConfiguration[]> {
         const configuredTasks = Array.from(this.tasksMap.values()).reduce((acc, labelConfigMap) => acc.concat(Array.from(labelConfigMap.values())), [] as TaskConfiguration[]);
         const detectedTasksAsConfigured: TaskConfiguration[] = [];
-        for (const [rootFolder, customizations] of Array.from(this.taskCustomizationMap.entries())) {
+        for (const [, customizations] of Array.from(this.taskCustomizationMap.entries())) {
             for (const cus of customizations) {
-                // TODO: getTasksToCustomize() will ask all task providers to contribute tasks. Doing this in a loop is bad.
-                const detected = await this.providedTaskConfigurations.getTaskToCustomize(token, cus, rootFolder);
+                const detected = this.taskDefinitionRegistry.getDefinition(cus);
```

Comment:
Question to reviewers:
- Did `getTaskToCustomize` do anything else other than the incorrect filtering?
- There seemed to be some fixes related to `scopes` there...
  - https://github.com/eclipse-theia/theia/blob/92a7d05c35b43a72560a514d51d1a65be3ff5899/packages/task/src/browser/provided-task-configurations.ts#L123

Reply:
I won't have time to look at this in-depth today, but getTasksToCustomize() did invoke "refreshTasks()" indirectly: so you no longer are asking the task providers to provide tasks anymore. But we need to do that once per user action (the "token" is to prevent providers being asked repeatedly.

Reply:
I will look into this again

Reply:
As far as I can see:
- `getTasksToCustomize()` was used to find the `detectedTask
- and then merge it with the customization in `getTasks()` and add the merged task to the results.
- However, it seems the `taskCustomizationMap` already has entries of `TaskConfiguration` (with _scope/label) so this merge should be a NOOP.

Although this merge may have meaning from a mutability perspective due to returning a **new** object.

Reply:
Can you guarantee that "refreshTasks" has been called at least once before this method is invoked? Because we need to ask task providers once per user interaction what their tasks are. They can be completely different for each execution of a "Run Task..." command.

Reply:
I cannot guarantee that, but we can make `refreshTasks` public and invoke it at the start of the function to
be on the safe side...

Reply:
Calling refreshTasks() with a token is a no-op if it has been called before with the same token, so this should be safe.